sway/desktop/render.c

```diff
@@ -986,6 +986,36 @@ static void render_seatops(struct sway_output *output,
 	}
 }
 
+static void render_im_popups(struct sway_output *output,
+		pixman_region32_t *damage) {
+	struct sway_seat *seat;
+	wl_list_for_each(seat, &server.input->seats, link) {
+		struct sway_input_popup *popup = seat->im_relay.input_popup;
+		if (popup == NULL || popup->popup_surface == NULL
+				|| !popup->popup_surface->mapped) {
+			continue;
+		}
+		struct render_data data = {
+			.damage = damage,
+			.alpha = 1.0f,
+		};
+		struct sway_view *view = view_from_wlr_surface(
+			popup->text_input->input->focused_surface);
```

Comment:
I encountered NULL popup->text_input dereference at here when I insert null check at [here](https://github.com/tadeokondrak/wlroots/commit/8960e09ae207e106603b0e7db16e3876881e4857#diff-d68541b5302b73ea493573e636e082fccb19a59399139f7adf1eca439742625cR458) and activate fcitx5.

Reply:
This seems to be a problem with this PR. It doesn't send frame callback events, probably due to the surface damage being incorrect. I'll keep testing with fcitx5.

Reply:
Hi, it works with fcitx5 now for me.